src/jaegertracing/TracerFactoryTest.cpp

```diff
@@ -65,12 +67,44 @@ TEST(TracerFactory, testValidConfig)
         "refreshInterval": 60
     }
   })";
+    testutils::EnvVariable::setEnv("JAEGER_SERVICE_NAME", "");
     TracerFactory tracerFactory;
     std::string errorMessage;
     auto tracerMaybe = tracerFactory.MakeTracer(config, errorMessage);
     ASSERT_EQ(errorMessage, "");
     ASSERT_TRUE(tracerMaybe);
 }
+
+TEST(TracerFactory, testEnvironmentConfig)
+{
+    const char* config = "";
+    testutils::EnvVariable::setEnv("JAEGER_SERVICE_NAME", "AService");
+    TracerFactory tracerFactory;
+    std::string errorMessage;
+    auto tracerMaybe = tracerFactory.MakeTracer(config, errorMessage);
+    ASSERT_EQ(errorMessage, "");
+    ASSERT_TRUE(tracerMaybe);
+}
+
+TEST(TracerFactory, testConfigAndEnvironment)
+{
+    testutils::EnvVariable::setEnv("JAEGER_SERVICE_NAME", "");
+    const char* config = R"(
+  {
+    "service_name": "test"
+  })";
+    testutils::EnvVariable::setEnv("JAEGER_SERVICE_NAME", "AService");
+    TracerFactory tracerFactory;
+    std::string errorMessage;
+    auto tracerMaybe = tracerFactory.MakeTracer(config, errorMessage);
+    ASSERT_EQ(errorMessage, "");
+    ASSERT_TRUE(tracerMaybe);
+
+    auto tracer = tracerMaybe.value();
+    const auto jaegerTracer = std::dynamic_pointer_cast<Tracer>(tracer);
```

Comment:
@yurishkuro any hints how I could the jaeger tracer instance here instead of the opentracing::tracer?

The dynamic pointer cast seems not to work, but I don't understand why :-(

Reply:
not really, but maybe @mdouaihy could suggest something

Reply:
@mdouaihy I tried:

```cpp
TEST(TracerFactory, testConfigAndEnvironment)
{
    testutils::EnvVariable::setEnv("JAEGER_SERVICE_NAME", "");
    const char* config = R"(
  {
    "service_name": "test"
  })";
    testutils::EnvVariable::setEnv("JAEGER_SERVICE_NAME", "AService");
    TracerFactory tracerFactory;
    std::string errorMessage;
    auto tracerMaybe = tracerFactory.MakeTracer(config, errorMessage);
    ASSERT_EQ(errorMessage, "");
    ASSERT_TRUE(tracerMaybe);

    const auto tracer =
        std::static_pointer_cast<jaegertracing::Tracer>(tracerMaybe.value());

    ASSERT_EQ(std::string("test"), tracer->serviceName());
}
```

but this segfaults. Any pointers?

Reply:
this is failing because the the returned tracer is not a Jaeger Tracer but a NoopTracer.
the reason is the environment variable JAEGER_DISABLED being set to true in ConfigTest.
I think that we should clear the set variables in those tests at the end of the test to avoid polluting other tests executions.

Reply:
Also note that you'll have to change the configuration of the Sampler and the Reporter to avoid trying to reach a remote address that doesn't exist.